### PR TITLE
feat(route53): support CNAME records

### PR DESCRIPTION
**Pull request by eladb**
Adds support for CNAME records via the `CnameRecord` construct.

Misc:

- `TXTRecord` was renamed to `TxtRecord`.
- `hostedZoneNameServers` attribute added to IHostedZone
- Made `HostedZone` a concrete (non-abstract) class so it will be
  compatible with how CloudFormation represents this resource,
  but left PublicHostedZone and PrivateHostedZone to allow
  a more strongly-typed experience if you like.

Credits for original PR (closes #1420): @MikeBild

BREAKING CHANGE: The `route53.TXTRecord` class was renamed to `route53.TxtRecord`.

----

### Pull Request Checklist

* [x] Testing
  - Unit test added
  - __CLI change?:__ manually run [integration tests](https://github.com/awslabs/aws-cdk/blob/master/packages/aws-cdk/integ-tests/test.sh) and paste output as a PR comment
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [x] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
* [x] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [x] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: packages/@aws-cdk/aws-route53/lib/records/cname.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  constructor(scope: Construct, id: string, props: CnameRecordProps) {
+    super(scope, id);
+
+    const ttl = props.ttl === undefined ? 1800 : props.ttl;
+
+    new CfnRecordSet(this, 'Resource', {
+      hostedZoneId: props.zone.hostedZoneId,
+      name: determineFullyQualifiedDomainName(props.recordName, props.zone),
+      type: 'CNAME',
+      resourceRecords: [ props.recordValue ],
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-route53/lib/records/cname.ts]: Why no support for multiple record values?

**Comment by MikeBild** [comment]
Very well! 👍

File: packages/@aws-cdk/aws-route53/lib/records/cname.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { Construct } from '@aws-cdk/cdk';
+import { IHostedZone } from '../hosted-zone-ref';
+import { CfnRecordSet } from '../route53.generated';
+import { determineFullyQualifiedDomainName } from './_util';
+
+export interface CnameRecordProps {
+  zone: IHostedZone;
+  recordName: string;
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-route53/lib/records/cname.ts]: Docstrings

File: packages/@aws-cdk/aws-route53/lib/records/cname.ts (hunk 1 above)
Comment (eladb) [reply packages/@aws-cdk/aws-route53/lib/records/cname.ts]: From the console:

```
[Duplicate Resource Record: <name>, RRSet of type CNAME with DNS name www.foo. does not contain exactly one resource record.]
```

File: packages/@aws-cdk/aws-route53/lib/records/cname.ts (hunk 2 above)
Comment (eladb) [reply packages/@aws-cdk/aws-route53/lib/records/cname.ts]: done. thanks

**Pull request merged**